pkg/utils/apiresources_test.go

```diff
@@ -228,144 +228,196 @@ func TestResourceConfigGroupParse(t *testing.T) {
-					Group:   "",
-					Version: "v1",
-					Kind:    "Node",
-				},
-				{
-					Group:   "",
-					Version: "v1",
-					Kind:    "Pod",
-				},
-				{
-					Group:   "authorization.k8s.io",
-					Version: "v1",
-					Kind:    "SelfSubjectRulesReview",
-				},
-				{
-					Group:   "apps",
-					Version: "v1",
-					Kind:    "HealthState",
-				},
-			},
-			enabled: []schema.GroupVersionKind{
-				{
-					Group:   "",
-					Version: "v1",
-					Kind:    "Ingress",
-				},
-				{
-					Group:   "apps",
-					Version: "v1beta1",
-					Kind:    "IngressClass",
-				},
-				{
-					Group:   "authorization.k8s.io",
-					Version: "v1",
-					Kind:    "LocalSubjectAccessReview",
-				},
-			},
+			Group:   "fleet.azure.com",
+			Version: "v1beta1",
+			Kind:    "MemberCluster",
+		},
+		{
+			Group:   "fleet.azure.com",
+			Version: "v1alpha1",
+			Kind:    "MemberCluster",
+		},
+		{
+			Group:   "events.k8s.io",
+			Version: "v1beta1",
+			Kind:    "Event",
 		},
 	}
-	for _, test := range tests {
-		r := NewDisabledResourceConfig()
-		if err := r.Parse(test.input); err != nil {
-			t.Fatalf("Unexpected error: %v", err)
-		}
-		for i, o := range test.disabled {
-			ok := r.IsResourceDisabled(o)
-			if !ok {
-				t.Errorf("%d: unexpected error: %v", i, o)
+
+	resourcesNotInDefaultResourcesList := []schema.GroupVersionKind{
+		{
+			Group:   "",
+			Version: "v1",
+			Kind:    "Namespace",
+		},
+		{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    "Deployment",
+		},
+		{
+			Group:   "",
+			Version: "v1",
+			Kind:    "Event",
+		},
+	}
+
+	tests := map[string]struct {
+		isAllowList bool
+		disabled    []schema.GroupVersionKind
+		enabled     []schema.GroupVersionKind
+	}{
+		"default disabled list": {
+			isAllowList: false,
+			disabled:    resourcesInDefaultDisabledList,
+			enabled:     resourcesNotInDefaultResourcesList,
+		},
+		"default enabled list": {
+			isAllowList: true,
+			disabled:    append(resourcesNotInDefaultResourcesList, resourcesInDefaultDisabledList...),
+			enabled:     []schema.GroupVersionKind{},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			r := NewResourceConfig(test.isAllowList)
+			if err := r.Parse(""); err != nil {
+				t.Fatalf("Parse() returned error: %v", err)
 			}
-		}
-		for i, o := range test.enabled {
-			ok := r.IsResourceDisabled(o)
-			if ok {
-				t.Errorf("%d: unexpected error: %v", i, o)
+			for i, o := range test.disabled {
+				if ok := r.IsResourceDisabled(o); !ok {
+					t.Errorf("%d: expected resource to be disabled : %v", i, o)
+				}
 			}
-		}
+
+			for i, o := range test.enabled {
+				if ok := r.IsResourceDisabled(o); ok {
+					t.Errorf("%d: expected resource to be enabled : %v", i, o)
+				}
+			}
+		})
 	}
 }
 
-func TestDefaultDisabledResourceConfigGroupVersionKindParse(t *testing.T) {
+func TestResourceConfigIsEmpty(t *testing.T) {
 	tests := []struct {
-		disabled []schema.GroupVersionKind
-		enabled  []schema.GroupVersionKind
+		input string
```

Comment:
could you please add a test name?

`In most tests, your test failure messages should include the function inputs if they are short. If the relevant properties of the inputs are not obvious (for example, because the inputs are large or opaque), you should name your test cases with a description of what’s being tested, and print the description as part of your error message.`

https://go.dev/wiki/TestComments#identify-the-input